volestipy/volestipy/volestipy.pyx

```diff
@@ -115,3 +126,86 @@ cdef class HPolytope:
       return self._A.shape[1]
+
+
+
+
+# build the low_dim_polytope_cpp class
+cdef class low_dim_HPolytope:
+
+   cdef lowDimHPolytopeCPP low_dim_polytope_cpp
+   cdef double[:,::1] _A
+   cdef double[::1] _b
+   cdef double [:,::1] _Aeq
+   cdef double[::1] _beq
+
+# set the specs of the class
+   def __cinit__(self, double[:,::1] A, double[::1] b, double[:,::1] Aeq, double[::1] beq):
+      self._A = A
+      self._b = b
+      self._Aeq = Aeq
+      self._beq = beq
+      n_rows_of_A, n_cols_of_A = A.shape[0], A.shape[1] 
+      n_row_of_Aeq, n_cols_of_Aeq = Aeq.shape[0], Aeq.shape[1] 
+      
+      self.low_dim_polytope_cpp = lowDimHPolytopeCPP(&A[0,0], &b[0], &Aeq[0,0], &beq[0], n_rows_of_A, n_cols_of_A, n_row_of_Aeq, n_cols_of_Aeq)
+
+   
+   # the get_full_dimensional_polytope() function(); that needs to run in case the user does not provide volestipy with a full dimensional polytope
+   def full_dimensiolal_polytope(self):
+      
+      # get dimensions of the initial S (Aeq) matrix
+      m = self._Aeq.shape[0]
+      n = self._Aeq.shape[1]
+      k = self._A.shape[0]
+      
+      # set the output variables
+      # the number of lines in the transpose N (columns in the actual matrix) are at least n-m; but we do not know their exact number
+      # so we initialize it with the maximum possible number of lines (n). the same is for the full A transpose matrix
+      # later, we will have to keep their actual dimension and remove these variables with the extra lines
+      cdef double[:,::1] N_extra_trans = np.zeros((n, n), dtype=np.float64, order="C")   
+      cdef double[::1] shift = np.zeros((n), dtype=np.float64, order="C")
+      cdef double[:,::1] A_full_extra_trans = np.zeros((n,k), dtype=np.float64, order="C")
+      cdef double[::1] b_full = np.zeros((k), dtype=np.float64, order="C")
+      
+      # we need to keep the final number of columns of the N / full_A matrices
+      cpdef int n_of_cols_in_N
+   
+      # call the C++ class to get the full_dimensional polytope
+      n_of_cols_in_N = self.low_dim_polytope_cpp.full_dimensiolal_polytope(&N_extra_trans[0,0], &shift[0], &A_full_extra_trans[0,0], &b_full[0])
+      
+      # get a matrix with exactly the number of lines and columns that N expands to and delete the one with the extra columns
+      N = np.zeros((n, n_of_cols_in_N), dtype=np.float64, order="C")
+      for i in range(n):
+         for j in range(n_of_cols_in_N):
+            N[i,j] = np.asarray(N_extra_trans[j,i])
+      del N_extra_trans
+         
+      # likewise, for the A matrix of the full dimensional polytope
+      A_full = np.zeros((k, n_of_cols_in_N), dtype=np.float64, order="C")
+      for i in range(k):
+         for j in range(n_of_cols_in_N):
+            A_full[i,j] = np.asarray(A_full_extra_trans[j,i])
+      del A_full_extra_trans
+      
+      # finally, we need to build an HP object for the full dumensional polytope we got
+      full_dimensional_polytope = HPolytope(A_full,b_full)
```

Comment:
Maybe
del A_full
del b_full ?

Reply:
If we remove ```b_full```, we should remove the ```A_full``` as well right?